datashape/coretypes.py

```diff
@@ -968,7 +968,35 @@ def __getitem__(self, types):
         if not isinstance(types, tuple):
             types = types,
 
-        return self(map(self._unpack_slice, types, range(len(types))))
+        return self(list(map(self._unpack_slice, types, range(len(types)))))
+
+
+def unify_name_types(names):
```

Comment:
could we conditionally define this function based on the version of python. This is just identity in python3 when str is unicode